lib/features/pokemon_overview/pokemon_overview_page.dart

```diff
@@ -20,15 +21,29 @@ class PokemonOverviewPage extends StatelessWidget {
         centerTitle: true,
         backgroundColor: primaryColor,
       ),
-      body: GridView.builder(
-        padding: const EdgeInsets.all(10.0),
-        itemCount: pokemons.length,
-        gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: pokemonColumnCount),
-        itemBuilder: (_, index) {
-          final pokemon = pokemons[index];
-          return PokemonCard(pokemon: pokemon);
+      body: pokemons.when(
+        loading: () => const Center(child: CircularProgressIndicator()),
+        error: (errorMessage) {
+          WidgetsBinding.instance.addPostFrameCallback((_) {
+            _showErrorMessageSnackbar(errorMessage, context: context);
+          });
+          return const Center(child: Text(noPokemonsAvailableLabel));
         },
+        (data) => GridView.builder(
+          padding: const EdgeInsets.all(10.0),
+          itemCount: data.length,
+          gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: pokemonColumnCount),
+          itemBuilder: (_, index) {
+            final pokemon = data[index];
+            return PokemonCard(pokemon: pokemon);
+          },
+        ),
       ),
     );
   }
+
+  void _showErrorMessageSnackbar(String? errorMessage, {required BuildContext context}) {
+    final snackBar = SnackBar(content: Text(errorMessage ??= emptyString));
```

Comment:
why use `null-aware assignment` here? just using null coalescing is already enough 🙂